ambry-commons/src/main/java/com/github/ambry/clustermap/HelixVcrUtil.java

```diff
@@ -266,6 +280,37 @@ private static boolean isDelayedRebalanceEnabled(String rebalancerClassName) {
     return (rebalancerClassName != null) && rebalancerClassName.endsWith(DELAYED_REBALANCER_CLASS_NAME);
   }
 
+  /**
+   * A method to verify resources and partitions in src cluster and dest cluster are same.
+   */
+  public static boolean isSrcDestSync(String srcZkString, String srcClusterName, String destZkString, String destClusterName) {
+
+    HelixAdmin srcAdmin = new ZKHelixAdmin(srcZkString);
+    Set<String> srcResources = new HashSet<>(srcAdmin.getResourcesInCluster(srcClusterName));
+    HelixAdmin destAdmin = new ZKHelixAdmin(destZkString);
+    Set<String> destResources = new HashSet<>(destAdmin.getResourcesInCluster(destClusterName));
+
+    for (String resource : srcResources) {
+      if (HelixVcrUtil.ignoreResourceKeyWords.stream().anyMatch(resource::contains)) {
+        System.out.println("Resource " + resource + " from src cluster is ignored");
+        continue;
+      }
+      if (destResources.contains(resource)) {
+        // check if every partition exist.
+        Set<String> srcPartitions = srcAdmin.getResourceIdealState(srcClusterName, resource).getPartitionSet();
+        Set<String> destPartitions = destAdmin.getResourceIdealState(destClusterName, resource).getPartitionSet();
+        for (String partition : srcPartitions) {
+          if (!destPartitions.contains(partition)) {
```

Comment:
I know this might not happen, but should we also add condition (destPatitions.size() == srcPartitions.size()) and another check for (destResources.size() = srcResources.size()).  -- this will check if somehow vcr cluster has extra partitions or resources that main cluster somehow doesn't have (can't think of any way this scenario can happen, so feel free to ignore this comment if you want)

Reply:
We can add the extra check when this happen. for now, I don't see this scenario could happen. 

If this happen, we need a vcr cluster rebuild, which is better to be done by manually cli.